### PR TITLE
Feature/encore actions

### PR DESCRIPTION
**Pull request by bishopbm1**
Added base action class with common code. Removed duplicate code from other actions. Added host_get_id action action to return the id of a Zabbix Host. Added host_update_status action to change the status of a Zabbix Host. Added maintenance_create_or_update action to create a maintenance window or update one if it already exists. Added maintenance_delete action to delete a maintenance window. Created new tests for all new actions and methods that were added to base action. Create base test class to inherit from. Updated readme and change log to incorporate new changes. Iterated the pack version.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -87,3 +87,6 @@ ENV/
 
 # Rope project settings
 .ropeproject
+
```
Comment (nmaludy) [inline .gitignore]: This is probably going to get rejected

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Change Log
 
+## 0.1.2
+
+- Added base action class with common code. Removed duplicate code from other actions
+- Added host_get_id action action to return the id of a Zabbix Host
+- Added host_update_status action to change the status of a Zabbix Host
+- Added maintenance_create_or_update action to create a maintenance window or update one if it already exists
+- Added maintenance_delete action to delete a maintenance window
+
```
Comment (nmaludy) [inline CHANGELOG.md]: Make sure to give yourself credit at the bottom of the list:

"Contributed by Brad Bishop (Encore Technologies)"

File: actions/host_get_id.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+name: host_get_id
+pack: zabbix
+runner_type: python-script
+description: Get the ID of a Zabbix Host
+enabled: true
+entry_point: host_get_id.py
+parameters:
+    host:
+        type: string
+        description: "Hostname of FQDN of the Zabbix Host"
```
Comment (nmaludy) [inline actions/host_get_id.yaml]: 'Hostname of FQDN of the Zabbix Host' doesn't make much sense to me?

Is it hostname (like first component) or the FQDN?

File: actions/host_update_status.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    def run(self, host=None, status=None):
+        """ Updates the status of a Zabbix Host. Status needs to be
+        1 or 0 for the call to succeed.
+        """
+        self.connect()
+
+        zabbix_host = self.find_host(host)
+
+        host_id = zabbix_host['hostid']
+
```
Comment (nmaludy) [inline actions/host_update_status.py]: This code is repeated a few times, can we move it into a common method in base?

File: actions/host_update_status.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+name: host_update_status
+pack: zabbix
+runner_type: python-script
+description: Update the status of a Zabbix Host
+enabled: true
+entry_point: host_update_status.py
+parameters:
+    host:
+        type: string
+        description: "Hostname of FQDN of the Zabbix Host"
```
Comment (nmaludy) [inline actions/host_update_status.yaml]: same hostname naming thing

File: actions/host_update_status.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        required: True
+    status:
+        type: integer
+        description: "Status to set the Zabbix Host to ex. 0 - monitored host 1 - unmonitored host"
```
Comment (nmaludy) [inline actions/host_update_status.yaml]: 'ex.' is the wrong terminology here, should be something like "valid values are: "

File: actions/maintenance_create_or_update.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            end_date=None):
+        """ Creates or updates a Zabbix maintenance window by looking
+        for the supplied maintenance_window_name and creating the mainenance window if it
+        does not exist or updating the mainenance window if it already exists.
+        """
+        self.connect()
+
+        zabbix_host = self.find_host(host)
+
+        host_id = zabbix_host['hostid']
```
Comment (nmaludy) [inline actions/maintenance_create_or_update.py]: copy/pasted code

File: actions/maintenance_create_or_update.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        if start_date is not None and end_date is not None:
+            local_tz = get_localzone()
+
+            start_local = datetime.strptime(start_date, "%Y-%m-%d %H:%M")
+            start_local = start_local.replace(tzinfo=local_tz)
+            start_time = int(start_local.strftime('%s'))
+
+            end_local = datetime.strptime(end_date, "%Y-%m-%d %H:%M")
+            end_local = end_local.replace(tzinfo=local_tz)
+            end_time = int(end_local.strftime('%s'))
```
Comment (nmaludy) [inline actions/maintenance_create_or_update.py:49]: copy/pasted code

**Review by nmaludy — CHANGES_REQUESTED** [review]

File: actions/maintenance_create_or_update.py (hunk 8 above)
Comment (bishopbm1) [reply actions/maintenance_create_or_update.py:49]: I think you were talking about the host_id code? If so it's fixed.

**Review by nmaludy — APPROVED** [review]

**Pull request merged**